src/web/rustdoc.rs

```diff
@@ -209,141 +209,157 @@ pub fn rustdoc_redirector_handler(req: &mut Request) -> IronResult<Response> {
     }
 
-    // docs have "rustdoc" prefix in database
+    // Add rustdoc prefix, name and version to the path for accessing the file stored in the database
     req_path.insert(0, "rustdoc");
-
-    // add crate name and version
     req_path.insert(1, &name);
     req_path.insert(2, &version);
 
-    let crate_details = cexpect!(CrateDetails::new(&conn, &name, &version));
-
-    // if visiting the full path to the default target, remove the target from the path
-    // expects a req_path that looks like `/rustdoc/:crate/:version[/:target]/.*`
-    if req_path[3] == crate_details.metadata.default_target {
-        return redirect(&name, &version, &req_path[4..]);
+    // Create the path to access the file from
+    let mut path = req_path.join("/");
+    if path.ends_with('/') {
+        req_path.pop(); // get rid of empty string
+        path.push_str("index.html");
+        req_path.push("index.html");
     }
 
-    let mut path = {
-        let mut path = req_path.join("/");
-        if path.ends_with('/') {
-            req_path.pop(); // get rid of empty string
-            path.push_str("index.html");
-            req_path.push("index.html");
-        }
-        path
-    };
+    // Attempt to load the file from the database
+    let file = if let Some(file) = File::from_path(&conn, &path) {
+        file
+    } else {
+        // If it fails, we try again with /index.html at the end
+        path.push_str("/index.html");
+        req_path.push("index.html");
 
-    let file = match File::from_path(&conn, &path) {
-        Some(f) => f,
-        None => {
-            // If it fails, we try again with /index.html at the end
-            path.push_str("/index.html");
-            req_path.push("index.html");
-            match File::from_path(&conn, &path) {
-                Some(f) => f,
-                None => return Err(IronError::new(Nope::ResourceNotFound, status::NotFound)),
-            }
-        }
+        File::from_path(&conn, &path)
+            .ok_or_else(|| IronError::new(Nope::ResourceNotFound, status::NotFound))?
     };
 
-    let req_path = req_path;
-
-    // serve file directly if it's not html
+    // Serve non-html files immediately
     if !path.ends_with(".html") {
         return Ok(file.serve());
     }
 
-    let mut content = RustdocPage::default();
-
     let file_content = ctry!(String::from_utf8(file.0.content));
-
+    // Extract the head and body of the rustdoc file so that we can insert it into our own html
     let (head, body, mut body_class) = ctry!(utils::extract_head_and_body(&file_content));
-    content.head = head;
-    content.body = body;
 
+    // Add the `rustdoc` classes to the html body
     if body_class.is_empty() {
         body_class = "rustdoc container-rustdoc".to_string();
     } else {
-        // rustdoc adds its own "rustdoc" class to the body
+         // rustdoc adds its own "rustdoc" class to the body 
         body_class.push_str(" container-rustdoc");
```

Comment:
I really need to preview my suggestions before hitting submit.

```suggestion
        // rustdoc adds its own "rustdoc" class to the body 
        body_class.push_str(" container-rustdoc");
```